core/lib/web3_decl/src/error.rs

```diff
@@ -66,8 +67,10 @@ pub fn is_retriable(err: &ClientError) -> bool {
         ClientError::Transport(_) | ClientError::RequestTimeout => true,
         ClientError::Call(err) => {
             // At least some RPC providers use "internal error" in case of the server being overloaded
+
             err.code() == ErrorCode::ServerIsBusy.code()
                 || err.code() == ErrorCode::InternalError.code()
+                || err.code() == SERVER_IS_SHUTTING_DOWN_ERROR
```

Comment:
3 is used as an error code for *all* `SubmitTransactionError`s, not just `ServerShuttingDown`, and some other error variants (`SerializationError`, `ProxyError`). Is this OK? If we want to distinguish `ServerShuttingDown` specifically, it'd be relatively easy to refactor the `Web3Error::SubmitTransactionError` variant (e.g., to make it include the JSON-RPC code / equivalent enum), so that `ServerShuttingDown` has a differing variant (standard `ErrorCode::ServerIsBusy` seems to match by semantics, and it's already handled above).

Reply:
that is the error i've found when disabled the server. I also haven't found why the number is 3

Reply:
Seems it'd be really better to distinguish these errors.

Reply:
Did something like this. Should be good